src/coreclr/jit/importer.cpp

```diff
@@ -7455,7 +7455,7 @@ int Compiler::impBoxPatternMatch(CORINFO_RESOLVED_TOKEN* pResolvedToken,
 
                 // See if the resolved tokens describe types that are equal.
                 const TypeCompareState compare =
-                    info.compCompHnd->compareTypesForEquality(unboxResolvedToken.hClass, pResolvedToken->hClass);
+                    info.compCompHnd->compareTypesForEquality(unboxResolvedToken.hClass, pResolvedToken->hClass, false);
```

Comment:
I wonder if we could just call `getTypeForPrimitiveValueClass` on the classes here. I guess it's not the exact same logic but maybe it is ok?

Reply:
Great idea! addressed 🙂